docs: fix RRR docs with azure (#1272)
redpanda-data/documentation: modules/manage/partials/authentication.adoc

==== Basic authentication

> NOTE: Redpanda recommends that you use TLS when enabling HTTP Basic Auth.

Reviewer: ```suggestion
NOTE: Redpanda Data recommends that you use TLS when enabling HTTP Basic Auth.
```

> NOTE: Redpanda recommends that you use TLS when HTTP Basic Auth is enabled.

Reviewer 2: ```suggestion
NOTE: Redpanda recommends that you use TLS when enabling HTTP Basic Auth.
```

NOTE: If you enable OIDC authentication for the Admin API, you must also <<create-superusers,create a SCRAM superuser>> so that you can use `rpk` to create ACLs. `rpk` supports only basic authentication for the Admin API. See <<Authentication for the HTTP APIs>>.

You can enable OIDC for HTTP APIs in two ways:

> * **Cluster-wide**: Use the `http_authentication` cluster property to enable OIDC for all HTTP API listeners

Reviewer: consider linking to the property

> * **Per-listener**: Use the `authentication_method` property to enable OIDC for specific API listeners

Reviewer: ```suggestion
* **Per-listener**: Use the `authentication_method` cluster property to enable OIDC for specific API listeners
```

Reviewer: is authentication_metod a cluster property? Kafka property? broker?

Reviewer 2: It appears to be missing from our reference docs

Reviewer 2: I *think* it might be a broker property, because you use it to specify the authentication method for a broker. Does not appear in our Broker properties section. So our automation does not pull these "complex" properties?

Reviewer 2: So I cannot link to it because it does not exist in any of our reference docs

> To enable OIDC cluster-wide for all HTTP API listeners, include both `BASIC` and `OIDC` in the `http_authentication` cluster property list:

Reviewer 3: Do I actually need to set both options if I just want OIDC? Also this was the issue described in DOC-401 no? This mentions `BASIC` and `OIDC` whereas `BASIC` should probably be `http_basic` and `OIDC` => `oidc`? (unsure if it's case sensitive, but I'm looking at the below example).

Maybe I'm completely on the wrong and the different options (cluster-wide vs per-listener) have different values? In that case I'd be surprised :D

Reviewer 4: You MUST set the node config,, i.e., for SR:
```yaml
schema_registry:
  schema_registry_api:
    - address: 127.0.0.5
      port: 18081
      name: basic
      authentication_method: http_basic
```
You MUST enable OIDC:
```shell
rpk cluster config set http_authentication '["OIDC"]
```
You MAY also enable BASIC:
```shell
rpk cluster config set http_authentication '["BASIC","OIDC"]
```

Reviewer 3: Ah this is very confusing, why is `authentication_method: http_basic` in the node config required / what does it do? 

The `http_basic` there is then **not** the same as the basic in `rpk cluster config set http_authentication '["BASIC","OIDC"]` I guess. (This is probably the source of confusion for me).

Reviewer 4: > Ah this is very confusing, why is authentication_method: http_basic in the node config required / what does it do?

It enables authentication for the listener.

> The http_basic there is then not the same as the basic in rpk cluster config set http_authentication '["BASIC","OIDC"] I guess. (This is probably the source of confusion for me).

Yes, it's confusing, which I pointed out at the time.

I also suggested a more informative name `http_authentication_mechanisms`, but that was considered too long.

Reviewer 2: Hoping the note I added clarifies this for users.

NOTE: If you enable OIDC authentication for the Admin API, you must also <<create-superusers,create a SCRAM superuser>> so that you can use `rpk` to create ACLs. `rpk` supports only basic authentication for the Admin API. See <<Authentication for the HTTP APIs>>.

> You can enable OIDC for HTTP APIs in two ways:

Reviewer 4: There are not two ways. Both of these things are required.

* **Cluster-wide**: Use the xref:reference/properties/cluster-properties.adoc#http_authentication[`http_authentication`] cluster property to enable OIDC for all HTTP API listeners

> * **Per-listener**: Use the `authentication_method` property to enable OIDC for specific API listeners

Reviewer 4: @paulohtb6 is fixing this, yes, it's a child of a broker property, specifically the HTTP listeners, `pandaproxy_api`, `schema_registry_api`.

rpk cluster config set http_authentication '["BASIC","OIDC"]'
----

Alternatively, you can enable OIDC for specific listeners by configuring the `authentication_method` property for individual API listeners. To enable both basic authentication and OIDC for specific listeners, set `authentication_method` to a list including both `http_basic` and `oidc`. For example, in `redpanda.yaml`, enter:

[,yaml,lines=5+10]
----
pandaproxy:
  pandaproxy_api:
  - address: "localhost"
    port: 8082

>     authentication_method: ["http_basic", "oidc"]

Reviewer 4: This is invalid

schema_registry:
  schema_registry_api:
    address: "localhost"
    port: 8081

>     authentication_method: ["http_basic", "oidc"]

Reviewer 4: This is invalid

----

To enable only OIDC authentication (without basic authentication) for specific listeners, set `authentication_method` to `oidc`:

[,yaml,lines=5+10]
----
pandaproxy:
  pandaproxy_api:
  - address: "localhost"
    port: 8082

>     authentication_method: oidc

Reviewer 4: This is invalid

schema_registry:
  schema_registry_api:
    address: "localhost"
    port: 8081

>     authentication_method: oidc

Reviewer 4: This is invalid.

rpk cluster config set http_authentication '["BASIC","OIDC"]'
----

> NOTE: Valid values for the cluster configuration `http_authentication` are `BASIC` and `OIDC`. Note that the value `BASIC` is different from the `http_basic` value, which enables authentication for the listener using the broker configuration `authentication_method`.

Reviewer: ```suggestion
NOTE: Valid values for the cluster configuration `http_authentication` are `BASIC` and `OIDC`. Note that the value `BASIC` is different from the `http_basic` value, which enables authentication for the listener using the broker configuration property `authentication_method`.
```

we should be consistent throughout the page if we want to say:
- broker configuration property
- broker property
- broker configuration

I vote for `broker configuration property`
